src/connectionbenchmark/benchmarktaskping.cpp

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "benchmarktask.h"
+#include "benchmarktaskping.h"
+#include "constants.h"
+#include "leakdetector.h"
+#include "logger.h"
+#include "mozillavpn.h"
+#include "networkrequest.h"
+#include "taskscheduler.h"
+
+#include <QTimer>
+
+namespace {
+Logger logger(LOG_MAIN, "BenchmarkTaskPing");
+}
+
+BenchmarkTaskPing::BenchmarkTaskPing()
+    : BenchmarkTask(Constants::BENCHMARK_MAX_DURATION_PING) {
+  MVPN_COUNT_CTOR(BenchmarkTaskPing);
+}
+
+BenchmarkTaskPing::~BenchmarkTaskPing() { MVPN_COUNT_DTOR(BenchmarkTaskPing); }
+
+void BenchmarkTaskPing::runInternal() {
+  logger.debug() << "Run ping benchmark";
+
+  connect(this, &BenchmarkTask::stateChanged, this,
+          &BenchmarkTaskPing::handleState);
+}
+
+void BenchmarkTaskPing::handleState(BenchmarkTask::State state) {
+  logger.debug() << "Handle state" << state;
+
+  if (state == BenchmarkTask::StateActive) {
+    connect(MozillaVPN::instance()->connectionHealth(),
+            &ConnectionHealth::pingChanged, this, [&] {
+              logger.debug() << "Ping changed";
+
+              m_pingLatencyAcc +=
+                  MozillaVPN::instance()->connectionHealth()->latency();
+              m_numOfPingSamples++;
+            });
+  } else if (state == BenchmarkTask::StateInactive) {
+    quint64 m_pingLatency = (int)(m_pingLatencyAcc / m_numOfPingSamples + 0.5);
```

Comment:
3 things:
1. why 0.5?
2. can you write it in this way: `(m_pingLatencyAcc / m_numOfPingSamples) + 0.5` - same result, just easier to read
3. `static_cast<int>(...)`

Reply:
plus... why int if you want a quin64?

Reply:
Casting `int` always rounds the value to the floor and this makes sure it is ”rounding” the result as I would expect it to if the decimal point is `>= 0.5`. So that for example: `1.8` → `2` but `1.3` → `1`. This might be not necessary if we follow Owens suggestion to directly take the ping value.